### ISSUE TITLE
updatePathway not working

### ISSUE THREAD

**Issue by ariutta** [wikipathways/wikipathways-api-client-js#1]
Attempting to run the [updatePathway test](https://github.com/wikipathways/wikipathways-api-client-js/blob/master/test/manual-tests/update-pathway.js) produces this error:

```
WS FAULT exception: [Sender] Revision out of date: your GPML code originates from an old revision. This means somebody else modified the pathway since you downloaded it. Please apply your changes on the newest version
```

The code for updating the pathway is [here](https://github.com/wikipathways/wikipathways-api-client-js/blob/ff6adb86f0b7469e923aee59ced3c174e82ab5d1/lib/main.js#L230-L280).

**Comment by ariutta** [comment]
Tried both 79997 and 79998. Got the same result.

**Comment by mkutmon** [comment]
Ok. I will try to give it a try tomorrow. Can you try it with using 0?
0 says that it should use the latest revision. Just to verify?

**Comment by ariutta** [comment]
I think I remember I trying ``0`` earlier and getting the same result.

But now I'm getting a 404 for this URL: ``http://webservice.wikipathways.org/updatePathway``, so I can't try it.

What should the API URL be for updating a pathways at pvjs.wikipathways.org? It's currently ``http://pvjs.wikipathways.org/wpi/webservicetest/updatePathway``, as set [here]( https://github.com/wikipathways/wikipathways-api-client-js/blob/a114e1773a3d2bab5de98e45022e277585d27261/lib/main.js#L44) and [here](https://github.com/wikipathways/wikipathways-api-client-js/blob/a114e1773a3d2bab5de98e45022e277585d27261/lib/main.js#L192).

**Comment by ariutta** [comment]
You can also try out this [demo/test](http://wikipathways.github.io/wikipathways-api-client-js/update-pathway.html). If you're developing locally, you can view it at http://localhost:3000/test/update-pathway.html

**Comment by AlexanderPico** [comment]
Ok. We are back to where we were... Getting an "originates from an old revision" error. From what I understand, the current revision that should be passed in to updatePathway(), where it is then incremented by one here (https://github.com/wikipathways/wikipathways-api-client-java/blob/master/org.wikipathways.client/src/org/pathvisio/wikipathways/webservice/WikiPathwaysRESTBindingStub.java#L540). 

So, according to both the pathway page for WP4 and this webservice result:
http://pvjs.wikipathways.org/wpi/webservicetest/index.php?method=getPathwayInfo&pwId=WP4
the current revision is 80113.

But, when we make this call:
http://pvjs.wikipathways.org/wpi/webservicetest/index.php?method=updatePathway&pwId=WP4&revision=80113&description=<<description>>&gpml=<<gpml>>
we get this error:
"Revision out of date: your GPML code originates from an old revision. This means somebody else modified the pathway since you downloaded it. Please apply your changes on the newest version"

I've also tried using an incremented revision, a revision = "0", and the revision from the live site. Same error each time.

I suspect that the error is either based on:
1) Improper passing of variable
2) Check against some other source of pathway information
3) A "red herring" and due to something else entirely.

I'm currently pursuing #3. If anyone has suggestions, I'd love to hear them!

**Comment by ariutta** [comment]
Fixed in commit e105e71fda11811962ac8bd339ecd6a83856e505
